scripts/python/PyAutoscoper/connect.py

```diff
@@ -54,14 +87,17 @@ def _send_command(self, command, *args):
         b = bytearray()
         b.append(command)
         b.extend(packed_data)
-        self.socket.sendall(b)
-        response = self._wait_for_server()
+        try:
+            self.socket.sendall(b)
+            response = self._wait_for_server()
+        except socket.error as e:
+            raise AutoscoperConnectionError(e)
         if response[0] != command:
             self.closeConnection()
-            raise Exception(f"Server error: received {response[0]}, expected {command}")
+            raise AutoscoperServerError(command, response[0])
```

Comment:
```suggestion
            raise AutoscoperServerError(f"received {response[0]}, expected {command}")
```

Since ultimately we may also associate message send back by the server, we should not prescribe one set of parameter.